src/mlpack/tests/main_tests/lsh_test.cpp

```diff
@@ -0,0 +1,357 @@
+ */
+BOOST_AUTO_TEST_CASE(LSHDiffProjectionsTest)
+{
+  arma::mat reference = arma::randu<arma::mat>(5, 100);
+
+  SetInputParam("reference", reference);
+  SetInputParam("k", (int) 6);
+
+  mlpackMain();
+
+  arma::Mat<size_t> neighbors = CLI::GetParam<arma::Mat<size_t>>("neighbors");
+  arma::mat distances = CLI::GetParam<arma::mat>("distances");
+
+  bindings::tests::CleanMemory();
+
+  // Train model using projections equals to 30.
+
+  SetInputParam("reference", std::move(reference));
+  SetInputParam("k", (int) 6);
+  SetInputParam("projections", (int) 30);
+
+  mlpackMain();
+
+  // Check that initial outputs and final outputs using two models are
+  // different.
+  BOOST_REQUIRE_LT(arma::accu(neighbors ==
+      CLI::GetParam<arma::Mat<size_t>>("neighbors")), neighbors.n_elem);
+  BOOST_REQUIRE_LT(arma::accu(distances ==
+      CLI::GetParam<arma::mat>("distances")), distances.n_elem);
+}
+
+/**
+ * Check learning process using different hash_width.
+ */
+BOOST_AUTO_TEST_CASE(LSHDiffHashWidthTest)
+{
+  arma::mat reference = arma::randu<arma::mat>(5, 100);
+
+  SetInputParam("reference", reference);
+  SetInputParam("k", (int) 6);
+
+  mlpackMain();
+
+  arma::Mat<size_t> neighbors = CLI::GetParam<arma::Mat<size_t>>("neighbors");
+  arma::mat distances = CLI::GetParam<arma::mat>("distances");
+
+  bindings::tests::CleanMemory();
+
+  // Train model using hash_width equals to 0.5.
+
+  SetInputParam("reference", std::move(reference));
+  SetInputParam("k", (int) 6);
+  SetInputParam("hash_width", (double) 0.5);
+
+  mlpackMain();
+
+  // Check that initial outputs and final outputs using two models are
+  // different.
+  BOOST_REQUIRE_LT(arma::accu(neighbors ==
+      CLI::GetParam<arma::Mat<size_t>>("neighbors")), neighbors.n_elem);
+  BOOST_REQUIRE_LT(arma::accu(distances ==
+      CLI::GetParam<arma::mat>("distances")), distances.n_elem);
+}
+
+/**
+ * Check learning process using different num_probes.
+ */
+BOOST_AUTO_TEST_CASE(LSHDiffNumProbesTest)
+{
+  arma::mat reference = arma::randu<arma::mat>(5, 100);
+
+  SetInputParam("reference", reference);
+  SetInputParam("k", (int) 6);
+
+  mlpackMain();
+
+  arma::Mat<size_t> neighbors = CLI::GetParam<arma::Mat<size_t>>("neighbors");
+  arma::mat distances = CLI::GetParam<arma::mat>("distances");
+
+  bindings::tests::CleanMemory();
+
+  // Train model using num_probes equals to 5.
+
+  SetInputParam("reference", std::move(reference));
+  SetInputParam("k", (int) 6);
+  SetInputParam("num_probes", (int) 5);
+
+  mlpackMain();
+
+  // Check that initial outputs and final outputs using two models are
+  // different.
+  BOOST_REQUIRE_LT(arma::accu(neighbors ==
+      CLI::GetParam<arma::Mat<size_t>>("neighbors")), neighbors.n_elem);
+  BOOST_REQUIRE_LT(arma::accu(distances ==
+      CLI::GetParam<arma::mat>("distances")), distances.n_elem);
+}
+
+/**
+ * Check learning process using different second_hash_size.
+ */
+BOOST_AUTO_TEST_CASE(LSHDiffSecondHashSizeTest)
```

Comment:
Since LSH builds random models, it might be best here to specify `mlpack::math::FixedRandomSeed()` before each `mlpackMain()` call.  Otherwise the test might pass even if `second_hash_size` is being ignored.  You can see `cf_test.cpp` for examples of the `FixedRandomSeed()` function in use.